### PR TITLE
Adds a locking system for spacepods.

### PR DESCRIPTION
**Pull request by MadmanMartian**
Now you can lock your pods!

:cl:
 * rscadd: Adds an optional locking module for spacepods.

### CODE REVIEW

**Comment by Kurfursten** [comment]
You might wanna add a little more description here in the PR

This might be dopey but until I read the code I thought it was a "lock on" module, e.g.: for weapon systems

**Comment by ghost** [comment]
How do you unlock a pod if you do not have the key?

I hope it won't be another mech DNA lock system where the pod gets bricked if owner dies.

**Comment by MadmanMartian** [comment]
working on that now

 - Good old emag
 - Maybe some multitool stuff

**Comment by Kurfursten** [comment]
Imagine the wires for pods

We need a "fire primary weapon" wire. You lure sec into the pod bay and slam that remote signaler to tase them from behind out of your pod. Pulse wire to enable/disable passenger boarding to sneak into someone's pod as a stowaway. Etc. etc.

File: code/WorkInProgress/pomf/spacepods/equipment.dm [hunk 1]
```diff
@@ -124,3 +126,44 @@
...
+			to_chat(user, "<span class = 'notice'>You pair \the [P] with \the [src].</span>")
+			P.code = code
+		else
+			to_chat(user, "<span class = 'warning'>\The [P] is already codelocked.</span>")
+		return
+	..()
+
+/obj/item/device/pod_key
+	name = "pod key"
+	desc = "used in tandem with a pod locking system"
```
Comment (Kurfursten) [inline code/WorkInProgress/pomf/spacepods/equipment.dm]: Used, also period

File: code/modules/research/designs/spacepod/weapons.dm [hunk 2]
```diff
@@ -28,3 +28,23 @@
...
 	materials = list(MAT_IRON = 15000)
 	locked = 1
+
+/datum/design/pod_lock
+	name = "Spacepod Equipment (Toggle Lock)"
+	desc = "Allows for the construction of a spacepod mounted locking system."
+	id = "pod_lock"
+	build_type = PODFAB
+	build_path = /obj/item/device/spacepod_equipment/locking/lock
+	req_tech = list(Tc_MATERIALS = 2, Tc_BLUESPACE = 2)
```
Comment (Kurfursten) [inline code/modules/research/designs/spacepod/weapons.dm:38]: Weird tech requirement but not enough to be controversial. Data or engineering might seem more natural?

**Comment by MadmanMartian** [comment]
Update: Pod key can be hit by a multitool to set its code.

Hitting an incorrect pod key against a pod will tell you if its code has any of the correct values, and/or correct positioning

**Comment by Kurfursten** [comment]
Looks good, is this ready to go?

**Comment by MadmanMartian** [comment]
ye

File: code/WorkInProgress/pomf/spacepods/equipment.dm [hunk 3]
```diff
@@ -124,3 +126,74 @@
...
+			if(code == L.code)
+				L.toggle_lock()
+				return
+			var/list/our_code = string2charlist(num2text(code))
+			var/list/their_code = string2charlist(num2text(L.code))
+			var/found_values = 0
+			var/correct_positions = 0
+			for(var/i=1 to our_code.len)
+				var/char = our_code[i]
+				var/their_char = their_code[i]
```
Comment (ShiftyRail) [inline code/WorkInProgress/pomf/spacepods/equipment.dm:194]: Variable defined but not used !

Comment (MadmanMartian) [reply code/WorkInProgress/pomf/spacepods/equipment.dm:194]: ree

**Pull request merged**